### PR TITLE
Reference line tables

### LINKED ISSUE #134: Add a frequency table of events at each reference line.
Addressed in [`reference_line_tables`](https://github.com/RhoInc/clinical-timelines/tree/reference-line-tables).

# Test notes
* For each reference line a corresponding frequency table appears in the left column beneath the controls.
* The title of the frequency table matches the reference line tooltip.
* The frequency table breaks down the events that overlap the reference line by event type, and if the timelines are grouped, by the group variable.

Use settings below:  
```
{
  "groupings":[
    {"value_col":"SEX","label":"sex"}
  ],
  "reference_lines":[
    {"timepoint":"2015-05-01","label":"timepoint"}
  ]
}
```

### PR DESCRIPTION
**Pull request by samussiah**
pre-refactor

# Issues
Closes #134

### CODE REVIEW

File: src/callbacks/onLayout/augmentOtherControls/yAxisGrouping.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { select as d3select } from 'd3';
+
+export default function yAxisGrouping(select, d) {
+    const selected = d3select(select).select('option:checked');
+
+    //Update grouping settings.
```
Comment (jwildfire) [inline src/callbacks/onLayout/augmentOtherControls/yAxisGrouping.js]: Not urgent, but I threw in an issue to remove the Y-axis grouping control when no groups are specified - #153

File: src/util/merge.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+var hasOwnProperty = Object.prototype.hasOwnProperty;
```
Comment (jwildfire) [inline src/util/merge.js]: Going to trust you on this one for now :) 

Is this copied/pasted, or written from scratch?

File: src/callbacks/onResize/drawReferenceLines/drawReferenceTable.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+export default function drawReferenceTable(reference_line) {
+    //Filter data on events that overlap reference line.
+    reference_line.wide_data = this.filtered_wide_data.filter(
+        d =>
+            this.config.x_parseFormat.parse(d[this.config.st_col]) <=
+                this.config.x_parseFormat.parse(reference_line.timepoint) &&
+            this.config.x_parseFormat.parse(d[this.config.en_col]) >=
+                this.config.x_parseFormat.parse(reference_line.timepoint)
+    );
+
```
Comment (jwildfire) [inline src/callbacks/onResize/drawReferenceLines/drawReferenceTable.js]: Will this grab single day events that match the reference line date?

File: src/callbacks/onResize/drawReferenceLines/drawReferenceTable.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            row
+                .append('td')
+                .text(d.key)
+                .attr('class', d => d.class + (d.class === 'poe-lower-level' ? ' poe-indent' : ''));
+            row
+                .append('td')
+                .text(d.n)
+                .attr('class', d => d.class);
+        });
+}
```
Comment (jwildfire) [inline src/callbacks/onResize/drawReferenceLines/drawReferenceTable.js]: 👏  nice work here.

**Comment by dschwentker** [comment]
The reference line/frequency table is not working in IE 11. When I use the latest commit from this branch, nothing renders (Controls, chart, etc.) and I see this error:
"TypeError: Object doesn't support property or method 'assign'"

**Pull request merged**